lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -199,10 +216,14 @@ HostManager::runNetwork(llvm::StringRef networkName,
 
   executor_->run(
       networks_[networkName].dag.root.get(), std::move(context), currentRun,
-      [&activeRequest = this->activeRequestCount_, callback,
+      [this, callback,
        name = networkName.str()](RunIdentifierTy runID, llvm::Error err,
                                  std::unique_ptr<ExecutionContext> context) {
-        --activeRequest;
+        --activeRequestCount_;
+        auto it = networks_.find(name);
+        if (it != networks_.end()) {
```

Comment:
Don't you need a lock around this networks_ access?

Reply:
Good catch! Yes, we do.